ui/src/components/layout/layouting.ts

```diff
@@ -0,0 +1,66 @@
+import { Edge, Position, StraightEdge } from "reactflow"
+import { EipFlowNode} from "../../api/flow"
+import { Layout } from "../../api/flow"
+import dagre from "@dagrejs/dagre"
+
+const NODE_WIDTH = 128
+const NODE_HEIGHT = 128
+
+const graph = new dagre.graphlib.Graph()
+graph.setDefaultEdgeLabel(() => ({}))
+
+export const newFlowLayout = (
+  nodes: EipFlowNode[],
+  edges: Edge[],
+  orientation: Layout["orientation"],
+  density: Layout["density"]
```

Comment:
You can reduce the number of parameters passed into `newFlowLayout` by passing a single layout object instead, with orientation and density fields.